File: Arrays/Java/IntersectionOfTwoArrays2.java
// Leetcode Question : https://leetcode.com/problems/intersection-of-two-arrays-ii/
class Solution {
    public int[] intersect(int[] nums1, int[] nums2) {
        Arrays.sort(nums1);
        Arrays.sort(nums2);
      
        int m = nums1.length;
        int n = nums2.length;
        int i = 0, j = 0;
        ArrayList<Integer> ans = new ArrayList<>();
      
        while (i < m && j < n) {
            if (nums1[i] < nums2[j]) {
                i++;
            } else if (nums1[i] > nums2[j]) {
                j++;
            } else {
                ans.add(nums1[i]);
                i++;
                j++;
            }
        }

        // Convert ArrayList to an array of integers
        int[] result = new int[ans.size()];
        int index = 0;
        for (int num : ans) {
            result[index++] = num;
        }
        return result;
    }
}

File: Arrays/CPP/IntersectionOfTwoArrays2.cpp
// Leetcode Question : https://leetcode.com/problems/intersection-of-two-arrays-ii/
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        // Sort both arrays
        sort(nums1.begin(), nums1.end());
        sort(nums2.begin(), nums2.end());

        int m = nums1.size();
        int n = nums2.size();
        int i = 0, j = 0;
        vector<int> ans;

        // Find intersection
        while (i < m && j < n) {
            if (nums1[i] < nums2[j]) {
                i++;
            } else if (nums1[i] > nums2[j]) {
                j++;
            } else {
                ans.push_back(nums1[i]);
                i++;
                j++;
            }
        }

        return ans;
    }
};
